src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -74,9 +77,28 @@ static void updateAttributeLinks(EndpointId endpoint)
     BitFlags<PumpStatus> pumpStatus;
 
     // Get the current control- and operation modes
-    Attributes::ControlMode::Get(endpoint, &controlMode);
     Attributes::OperationMode::Get(endpoint, &operationMode);
-    Attributes::PumpStatus::Get(endpoint, &pumpStatus);
+
+    if (isControlModeAvailable)
+    {
+        Attributes::ControlMode::Get(endpoint, &controlMode);
+    }
+    else
+    {
+        // If controlMode attribute is not available, then use the default value
+        // of the effectiveControlMode attriute as the effectiveControlMode
+        // if this is not suitable, the application should override this value in
+        // the post attribute change callback for the operation mode attribute
+        const EmberAfAttributeMetadata * effectiveControlModeMetaData;
+        effectiveControlModeMetaData = GetAttributeMetadata(
```

Comment:
What guarantees this does not return null?

Reply:
The EffectiveControlMode mode attribute is mandatory and is not nullable, so I would assume it should be available?

Reply:
If it's mandatory and we assume that nothing is too misconfigured, then yes, this should be fine.